### ISSUE TITLE
Rules for setting learning rates

### ISSUE THREAD

**Issue by Franciszzj** [msracver/Deformable-ConvNets#82]
```
  lr: 0.0005
  lr_step: '4.83'
  warmup: true
  warmup_lr: 0.00005
  # typically we will use 4000 warmup step for single GPU on VOC
  warmup_step: 1000
```
If I change training dataset, how could I set lr_step, warmup_step and so on. 
```
    # decide learning rate
    base_lr = lr
    lr_factor = config.TRAIN.lr_factor
    lr_epoch = [float(epoch) for epoch in lr_step.split(',')]
    lr_epoch_diff = [epoch - begin_epoch for epoch in lr_epoch if epoch > begin_epoch]
    lr = base_lr * (lr_factor ** (len(lr_epoch) - len(lr_epoch_diff)))
    lr_iters = [int(epoch * len(roidb) / batch_size) for epoch in lr_epoch_diff]
    print('lr', lr, 'lr_epoch_diff', lr_epoch_diff, 'lr_iters', lr_iters)
    lr_scheduler = WarmupMultiFactorScheduler(lr_iters, lr_factor, config.TRAIN.warmup, 
    config.TRAIN.warmup_lr, config.TRAIN.warmup_step)
```
I viewed this code, can you explain in detail about each parameter?

**Comment by Franciszzj** [comment]
@YuwenXiong  Looking forward to your reply.

**Comment by Franciszzj** [comment]
‘len(roidb)’ means the number of instance. '4.83' means how many epoches to reduce lr.

**Comment by bfialkoff** [comment]
Can you please clarify what "'4.83' means how many epoches to reduce lr." means. Do you mean every 4.83 epochs reduce the learning rate (what would that even mean)? Please see this as well [https://github.com/msracver/Deformable-ConvNets/issues/248](url)
